Create logical volume for openshift tests.

dgoodwin opened the pull request:
Incoming extended tests will require an XFS filesystem mounted with gquota to
pass. Create a partition for /tmp/openshift where test data lands, including
the volume directory.

danmcp commented on lib/vagrant-openshift/action/install_origin_base_dependencies.rb:
```diff
@@ -215,6 +215,12 @@ def call(env)
   sudo sed -i "s,^DOCKER_STORAGE_OPTIONS=.*,DOCKER_STORAGE_OPTIONS='-s devicemapper --storage-opt dm.datadev=/dev/${VG}/docker-data --storage-opt dm.metadatadev=/dev/${VG}/docker-metadata'," /etc/sysconfig/docker-storage
+
+  sudo lvcreate -n volume-dir -l 100%FREE /dev/${VG}
+  sudo mkfs.xfs /dev/docker-vg/volume-dir
```
Need to use ${VG}

And can you pick a more meaningful name than volume-dir?

danmcp commented on lib/vagrant-openshift/action/install_origin_base_dependencies.rb:
```diff
@@ -215,6 +215,12 @@ def call(env)
+  sudo lvcreate -n volume-dir -l 100%FREE /dev/${VG}
+  sudo mkfs.xfs /dev/docker-vg/volume-dir
+  sudo mkdir -p /tmp/openshift/
+  sudo sh -c 'echo "/dev/docker-vg/volume-dir /tmp/openshift xfs gquota 1 1" >> /etc/fstab'
```
Need to use ${VG}

dgoodwin commented:
Updated now, I went with openshift-test-tmp, and replaced the two missing ${VG}'s. (my bad should have spotted that) Re-tested with the centos7 aws image.

[root@ip-172-18-5-200 centos]# df -h
Filesystem                                   Size  Used Avail Use% Mounted on
[snip]
/dev/mapper/docker--vg-openshift--test--tmp 1014M   33M  982M   4% /tmp/openshift

danmcp commented on lib/vagrant-openshift/action/install_origin_base_dependencies.rb:
```diff
@@ -215,6 +215,12 @@ def call(env)
   sudo lvcreate -n docker-metadata -l 50%FREE /dev/${VG}
   sudo sed -i "s,^DOCKER_STORAGE_OPTIONS=.*,DOCKER_STORAGE_OPTIONS='-s devicemapper --storage-opt dm.datadev=/dev/${VG}/docker-data --storage-opt dm.metadatadev=/dev/${VG}/docker-metadata'," /etc/sysconfig/docker-storage
+
+  sudo lvcreate -n openshift-test-tmp -l 100%FREE /dev/${VG}
```
Should xfs be in the name?   It seems more specific than just openshift test tmp.

dgoodwin replied:
It certainly can be, though we do mount it as the overall openshift test dir. 

How about openshift-xfs-tmp

danmcp replied:
@dgoodwin +1

dgoodwin commented:
Updated.

danmcp commented on lib/vagrant-openshift/action/install_origin_base_dependencies.rb:
```diff
@@ -215,6 +215,12 @@ def call(env)
+
+  sudo lvcreate -n openshift-xfs-tmp -l 100%FREE /dev/${VG}
+  sudo mkfs.xfs /dev/${VG}/openshift-xfs-tmp
+  sudo mkdir -p /tmp/openshift
```
Sorry I thought you were going to change the mnt point name as well.  Seems like we should.

dgoodwin replied:
At this point we'll be getting into territory I don't quite understand, I know the extended tests appear to run in /tmp/openshift so my plan was to push this onto the new xfs LV entirely, the change shouldn't affect any other test and will let this new test pass.

I assume you're thinking my new tests somehow re-initialize all the test infra in another location and initialize a master/node from there? I will need to do some digging to find out how to do this.

danmcp replied:
Ah.  Sorry I forgot this directory was already being used.  I was thinking you were adding a new dir only used by specific tests.   As it is now, I don't think this dir is going to be big enough to do everything /tmp/openshift was doing.  Do you know big it gets today?

dgoodwin replied:
In fairness, I probably could do this with a new dir, I do have to modify node-config for the test anyhow so it wouldn't be hard to point volumeDirectory at something else.

This is getting a bit more coupled, the test would have to assume a LV or a mount point exists, but it could definitely be done. 

It would also be nice to be able to reformat the volume after or before the test runs, as there is no reasonable way to delete an XFS quota from the filesystem after it's applied.

I have the tests running now, just a few minutes in and it's at 386M so indeed 1G is probably going to fail.

I'll consider this a sign to go with a separate volume directory for this particular test. I'll mount to /tmp/openshift/xfs-volume-dir. Would it be safe to trash it and format the fs each test? (I.e. assuming that it's safe to mess with an LV with this name) Getting pretty tightly coupled but I don't see any other way.

dgoodwin replied:
Do the extended tests in jenkins run on a separate VM every time? Or does one vm ever get re-used?

danmcp replied:
Separate VM every time.

dgoodwin replied:
Ok good, that's a lot less coupling there then. Proceeding with the above.

dgoodwin commented:
Updated and re-tested.
